test/types/community/collection/updateX.test-d.ts

```diff
@@ -431,9 +431,18 @@ export async function testPushWithId(): Promise<void> {
   expectAssignable<Promise<{ upsertedId: ObjectId | null }>>(collection.updateOne({}, {}));
   expectAssignable<Promise<{ upsertedId: ObjectId | null }>>(collection.updateMany({}, {}));
 
+  expectNotAssignable<Promise<{ upsertedId: number | null }>>(collection.updateOne({}, {}));
+  expectNotAssignable<Promise<{ upsertedId: number | null }>>(collection.updateMany({}, {}));
+
   const collectionWithSchema = {} as any as Collection<{ _id: number }>;
   expectAssignable<Promise<{ upsertedId: number | null }>>(
     collectionWithSchema.updateOne({ _id: 1234 }, {})
   );
   expectAssignable<Promise<{ upsertedId: number | null }>>(collectionWithSchema.updateMany({}, {}));
+  expectNotAssignable<Promise<{ upsertedId: ObjectId | null }>>(
+    collectionWithSchema.updateOne({ _id: 1234 }, {})
+  );
+  expectNotAssignable<Promise<{ upsertedId: ObjectId | null }>>(
```

Comment:
I think this one should be `number` no?

Reply:
No?  The schema mandates that _id is of type `number`, so we expect `upsertedId` must be `number | null` (not `ObjectId`)

Reply:
sorry I misread that